dumper/channelDataDumper.go

```diff
@@ -116,20 +120,23 @@ func processAndInsertProducts(db *sql.DB, writter *bufio.Writer) {
 	dumpAllTablesData(db, writter, schemaMetadata, startingTables, whereFilterClause)
 }
 
-func processAndInsertChannels(db *sql.DB, channelLabels []string, writter *bufio.Writer) DataDumper{
+func processAndInsertChannels(db *sql.DB, channelLabels []string, writter *bufio.Writer) []DataDumper {
 	schemaMetadata := schemareader.ReadTablesSchema(db, SoftwareChannelTableNames())
-
+	tableDumper := make([]DataDumper, 0)
```

Comment:
I think it would be easier to read if you create a second method for processing just one channel, and then have this one with will interate through the list of chanels and process one by one. It would be easier to read and maintain.